microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/project/JDTMicroProfileProject.java

```diff
@@ -45,23 +48,89 @@ public JDTMicroProfileProject(IJavaProject javaProject) {
 		configSources.add(new PropertiesConfigSource(MICROPROFILE_CONFIG_PROPERTIES_FILE, javaProject));
 	}
 
-	public String getProperty(String key, String defaultValue) {
-		for (IConfigSource configSource : configSources) {
-			String property = configSource.getProperty(key);
-			if (property != null) {
-				return property;
+	/**
+	 * Returns the value of this property or <code>defaultValue</code> if it is not
+	 * defined in this project
+	 *
+	 * @param propertyKey  the property to get with the profile included, in the
+	 *                     format used by microprofile-config.properties
+	 * @param defaultValue the value to return if the value for the property is not
+	 *                     defined in this project
+	 * @return the value of this property or <code>defaultValue</code> if it is not
+	 *         defined in this project
+	 */
+	public String getProperty(String propertyKey, String defaultValue) {
+		// Need to enforce the order that config files are looked at
+		// eg. prefer a Yaml config value over application.properties
+		IConfigSource configSource;
+		for (int i = 0; i < configSources.size(); i++) {
+			configSource = configSources.get(i);
+			String propertyValue = configSource.getProperty(propertyKey);
+			if (propertyValue != null) {
+				return propertyValue;
 			}
 		}
 		return defaultValue;
 	}
 
+	/**
+	 * Returns the value of this property or null if it is not defined in this
+	 * project
+	 *
+	 * @param propertyKey the property to get with the profile included, in the
+	 *                    format used by microprofile-config.properties
+	 * @return the value of this property or null if it is not defined in this
+	 *         project
+	 */
+	public String getProperty(String propertyKey) {
+		return getProperty(propertyKey, null);
+	}
+
 	public Integer getPropertyAsInteger(String key, Integer defaultValue) {
-		for (IConfigSource configSource : configSources) {
+		// Need to enforce the order that config files are looked at
+		// eg. prefer a Yaml config value over application.properties
+		IConfigSource configSource;
+		for (int i = 0; i < configSources.size(); i++) {

```

Comment:
for(IConfigSource configSource : configSources) {

Reply:
duplicate